src/core/init.ts

```diff
@@ -20,21 +20,37 @@ export function init(debug: boolean): void {
   initSDK();
 
   // Mount all components used in the project.
-  backButton.isSupported() && backButton.mount();
-  miniApp.mount();
-  themeParams.mount();
+  if (backButton.isSupported()) {
+    backButton.mount();
+  }
+
   initData.restore();
-  void viewport.mount().catch(e => {
-    console.error('Something went wrong mounting the viewport', e);
-  });
+
+  if (!miniApp.isMounted()) {
+    miniApp.mount();
+    miniApp.bindCssVars(); // should bind it here
+  }
+
+  if (!themeParams.isMounted()) {
+    themeParams.mount();
+  }
+
+  if (!viewport.isMounted() && !viewport.isMounting()) {
+    void viewport.mount().catch((e) => {
+      console.error("Something went wrong mounting the viewport", e);
+    });
+  }
```

Comment:
```suggestion
    void viewport.mount().then(() => viewport.bindCssVars()).catch((e) => {
      console.error("Something went wrong mounting the viewport", e);
    });
  }
```

Should bind after promise resolved